### PR TITLE
Remove the need for an initial page view.

### PR DESCRIPTION
**Pull request by bnorton**

### CODE REVIEW

**Comment by bnorton** [comment]
Note: I am pulling in upstream changes (from another integration) specifically around 'modernizing the test harness' commits in other repos.

**Comment by bnorton** [comment]
@ndhoule + @ Peripheral1994 This is ready to go! It includes small updates to not `assumePageview` and to not pass through an `anonymousId` as if it were the real `userId`

Thanks for having a look!

p.s. If there is someone else to loop in can you tag them here?

**Comment by bnorton** [comment]
@ndhoule + @Peripheral1994 This is ready to go! It includes small updates to not assumePageview and to not pass through an anonymousId as if it were the real userId

Thanks for having a look!

p.s. If there is someone else to loop in can you tag them here?

**Comment by bnorton** [comment]
Bump to @sperand-io @hankim813 @stevenmiller888 anything else needed here?

File: lib/index.js [hunk 1]
```diff
@@ -55,29 +53,10 @@ Chameleon.prototype.loaded = function() {
 Chameleon.prototype.identify = function(identify) {
   var options = identify.traits();
 
-  options.uid = options.id || identify.userId() || identify.anonymousId();
+  var uid = options.id || identify.userId() || null;
```
Comment (hankim813) [inline lib/index.js]: we can probably just do:

```
var traits = identify.traits();
// remove duplicate id
delete traits.id;

window.chmln.identify(identify.userId(), traits);
```

Comment (hankim813) [reply lib/index.js]: and note that `userId()` can return `undefined` for anonymous users

File: lib/index.js [hunk 2]
```diff
@@ -55,29 +53,10 @@ Chameleon.prototype.loaded = function() {
...
-};
-
-/**
- * Associate the current user with a group of users.
- *
- * @api public
- * @param {Facade} group
- */
-
-Chameleon.prototype.group = function(group) {
```
Comment (hankim813) [inline lib/index.js]: any reason why we are removing this? this will be a breaking change

File: lib/index.js [hunk 3]
```diff
@@ -89,15 +68,3 @@ Chameleon.prototype.group = function(group) {
...
   window.chmln.track(track.event(), track.properties());
 };
-
-/**
- * Change the user identifier after we know who they are.
- *
- * @param {Facade} alias
- */
-
-Chameleon.prototype.alias = function(alias) {
```
Comment (hankim813) [inline lib/index.js]: same here

**Review by hankim813 — CHANGES_REQUESTED** [review]
sorry for the delay @bnorton but some comments to resolve! thanks!

File: lib/index.js (hunk 2 above)
Comment (bnorton) [reply lib/index.js]: @hankim813 Those functions were no-ops on the backend for us -- I'll add them back with similar implementation on the frontend.

**Comment by bnorton** [comment]
@hankim813 the requested changes are in! -- thanks

File: lib/index.js [hunk 4]
```diff
@@ -69,15 +65,12 @@ Chameleon.prototype.identify = function(identify) {
...
-  var options = {};
-
-  each(function(value, key) {
-    options['group:' + key] = value;
-  }, group.traits());
+  var traits = group.traits();
+  delete traits.id;
 
-  options['group:id'] = group.groupId();
+  traits.uid = group.groupId();
```
Comment (hankim813) [inline lib/index.js]: ahh if you need to alias, we can write this entire function like this:

```
window.chmln.set({ company: group.traits({ id: 'uid' }) });
```

**Review by hankim813 — CHANGES_REQUESTED** [review]
lgtm, small nitpick. 

can you also add a release commit on top of this?

bump `package.json` version ... probably a patch? I'm interpreting this is mainly a bug fix + refactor?

then update `HISTORY.md` with convention
`$ git commit -m 'Release x.x.x'`

**Comment by hankim813** [comment]
@bnorton ahh so close! you forgot to bump the `version` in `package.json` to `2.0.1`!

**Comment by bnorton** [comment]
@hankim813 yikes you're right -- I force pushed to cover my mistake :)

**Review by hankim813 — APPROVED** [review]

**Comment by hankim813** [comment]
@bnorton deployed! might want to wait 10 min or so for cache to clear

**Pull request merged**